### PR TITLE
Account flow

### PR DESCRIPTION
**Pull request by JFATAJ**
@pelusodan pr for account flow, I have already merged the changes from mao's branch in here so we should have no conflicts.

### CODE REVIEW

**Review by pelusodan — CHANGES_REQUESTED** [review]
This looks good, but it's hard to tell what you worked on / added because there are so many files changed / so many commits. It was unfortunately a bit premature, as I have requested some changes from Mao on his side, which now means you need to update this branch once that is done. In the future, you should always work off the main branch, and then when you are ready to merge in updated changes from the main branch you can stash your changes, pull, then unstash. If you already have the branch open when you'd like to update with main, you can then do a git rebase main, and then force push to your branch which shows only your PRs. As I said, this looks good for now but its hard for me to tell what your changes are because there are so many files changed / commits. Once Mao's changes get pushed to master, we can merge this in. If you can, try doing a rebase and force push to see if you can have only your commits on this PR. You might need to manually add your commits to a new branch and start the PR over.

File: app/src/main/java/com/peluso/walletguru/ui/AddAccountFragment.kt [hunk 1]
```diff
@@ -1,48 +1,141 @@
...
-            //TODO: make this pop up dialog to add account
+            launchSubmitNewAccount()
         }
     }
 
-    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        viewModel.viewState.observe(viewLifecycleOwner, Observer {
-            handleViewState(it)
-        })
+    private fun launchSubmitNewAccount() {
```
Comment (pelusodan) [inline app/src/main/java/com/peluso/walletguru/ui/AddAccountFragment.kt:64]: this function should perform some sort of check which sees if there are any account types they are able to add. If I've added all the possible accounts, then this dialog should not open. Maybe then I see a toast message that says 'already added the maximum account types!'

Comment (JFATAJ) [reply app/src/main/java/com/peluso/walletguru/ui/AddAccountFragment.kt:64]: done

**Comment by JFATAJ** [comment]
Will create a separate pr after Mao merges his changes.

**Pull request closed without merging**